platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/StellarPaymentObserver.java

```diff
@@ -139,18 +153,33 @@ public void onEvent(OperationResponse operationResponse) {
             if (observedPayment != null) {
               try {
                 if (paymentObservingAccountsManager.lookupAndUpdate(observedPayment.getTo())) {
-                  final ObservedPayment finalObservedPayment = observedPayment;
-                  observers.forEach(observer -> observer.onReceived(finalObservedPayment));
+                  for (PaymentListener listener : observers) {
+                    listener.onReceived(observedPayment);
+                  }
                 }
+
                 if (paymentObservingAccountsManager.lookupAndUpdate(observedPayment.getFrom())
                     && !observedPayment.getTo().equals(observedPayment.getFrom())) {
                   final ObservedPayment finalObservedPayment = observedPayment;
                   observers.forEach(observer -> observer.onSent(finalObservedPayment));
                 }
+
+              } catch (EventPublishException ex) {
+                // restart the observer from where it stopped, in case the queue fails to
+                // publish the message.
+                Log.errorEx("Failed to send event to observer.", ex);
+                Log.info("Restarting the Stellar observer.");
+                restart();
+                return;
               } catch (Throwable t) {
```

Comment:
This can be dangerous when we try to shut down the service gracefully. We want to be able to interrupt a thread and leave the process to terminate gracefully. 

Can we check if the Thread is interrupted? And if interrupted, terminate the observer gracefully?

Reply:
Basically, `catch (throwable)` could try to reset a legit/graceful stop as if it was an error, right?

// TODO: check if `Thread.isInterrupted()`, or if we got an InterruptedException.

Reply:
Addressed in https://github.com/stellar/java-stellar-anchor-sdk/pull/607/commits/0b1c357bcf091ffb33f5a5bef4fef70c5877d2dc.